client/homebrew/editor/editor.jsx

```diff
@@ -162,7 +166,27 @@ const Editor = createClass({
 						if(match)
 							endCh = match.index+match[0].length;
 						codeMirror.markText({ line: lineNumber, ch: 0 }, { line: lineNumber, ch: endCh }, { className: 'block' });
+					};
+					
+					if(line.trimLeft().startsWith('{{')){
+						x += 1;
+						blockTypes.forEach(type=>{
+							if(line.includes(type)){   // todo: likely change to "starts with" rather than include to avoid overlapping issues
+								blockClass = type;
+							} 
+						});
+						if(blockClass === null){ blockClass = 'blockHighlight'}
+					}
+					if(x>0){
+						codeMirror.addLineClass(lineNumber, 'background', blockClass);
+					}
+					if(line.trimLeft().startsWith('}}')){
+						x -= 1;
+						blockClass = null; // todo:  need to switch back to previous class if nested within another div
```

Comment:
an array with push/pop is perfect for this type of tracking

Reply:
That's exactly what I was thinking. Thanks for mentioning it so now I can be sure before working through that

Reply:
I think it can probably be refactored to something more concise, but committed a fix for nested divs (and also excluded spans at the beginning of a line being included).  Did not fix that 'x' variable yet.